fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -185,127 +187,178 @@ protected static Tuple convertOrNull(@Nullable Key.Evaluated key) {
         }
     }
 
+    enum BuildType {
```

Comment:
It seems uncommon to define a enum just for being used in one method. If you'd like to eliminate it, I think you can inline
```
            if (buildType == BuildType.NO_BUILD) {
                 return AsyncUtil.DONE;
             }
             final boolean continuity = buildType == BuildType.CONTINUITY;
             return setIndexingTypeOrThrow(continuity)
                     .thenCompose(ignore -> buildIndexInternalAsync());
```
into the positions where your return the completed `BuildType`

Reply:
Oh, I thought it was more readable this way, but sure - I can change it.